opentelemetry-sdk/src/metrics/meter_provider.rs

```diff
@@ -217,42 +217,69 @@ impl fmt::Debug for MeterProviderBuilder {
+        SERVICE_NAME, TELEMETRY_SDK_LANGUAGE, TELEMETRY_SDK_NAME, TELEMETRY_SDK_VERSION,
+    };
     use crate::testing::metrics::metric_reader::TestMetricReader;
     use crate::Resource;
     use opentelemetry::global;
-    use opentelemetry::Key;
-    use opentelemetry::KeyValue;
+    use opentelemetry::{Key, KeyValue, Value};
     use std::env;
 
     #[test]
     fn test_meter_provider_resource() {
-        // If users didn't provide a resource and there isn't a env var set. Use default one.
-        let assert_service_name = |provider: super::SdkMeterProvider,
-                                   expect: Option<&'static str>| {
+        let assert_resource = |provider: &super::SdkMeterProvider,
+                               resource_key: &'static str,
+                               expect: Option<&'static str>| {
             assert_eq!(
                 provider.pipes.0[0]
                     .resource
-                    .get(Key::from_static_str("service.name"))
+                    .get(Key::from_static_str(resource_key))
                     .map(|v| v.to_string()),
                 expect.map(|s| s.to_string())
             );
         };
+        let assert_telemetry_resource = |provider: &super::SdkMeterProvider| {
+            assert_eq!(
+                provider.pipes.0[0]
+                    .resource
+                    .get(TELEMETRY_SDK_LANGUAGE.into()),
+                Some(Value::from("rust"))
+            );
+            assert_eq!(
+                provider.pipes.0[0].resource.get(TELEMETRY_SDK_NAME.into()),
+                Some(Value::from("opentelemetry"))
+            );
+            assert_eq!(
+                provider.pipes.0[0]
+                    .resource
+                    .get(TELEMETRY_SDK_VERSION.into()),
+                Some(Value::from(env!("CARGO_PKG_VERSION")))
+            );
+        };
+        // If users didn't provide a resource and there isn't a env var set. Use default one.
         let reader = TestMetricReader::new();
         let default_meter_provider = super::SdkMeterProvider::builder()
             .with_reader(reader)
             .build();
-        assert_service_name(default_meter_provider, Some("unknown_service"));
+        assert_resource(
+            &default_meter_provider,
+            SERVICE_NAME,
+            Some("unknown_service"),
+        );
+        assert_telemetry_resource(&default_meter_provider);
 
         // If user provided a resource, use that.
         let reader2 = TestMetricReader::new();
         let custom_meter_provider = super::SdkMeterProvider::builder()
             .with_reader(reader2)
             .with_resource(Resource::new(vec![KeyValue::new(
-                "service.name",
+                SERVICE_NAME,
                 "test_service",
             )]))
             .build();
-        assert_service_name(custom_meter_provider, Some("test_service"));
+        assert_resource(&custom_meter_provider, SERVICE_NAME, Some("test_service"));
```

Comment:
nit - as earlier, this test is redundant, and can be removed. As we are checking the length in the next test.

Reply:
In this case we want to assert the custom defined service_name value ("test_service"), but also that no other resources are defined (like the telemetry ones)